lib/models/DataPoint.dart

```diff
@@ -0,0 +1,33 @@
+class DataPoint {
+  DateTime date = DateTime.now();
+  int dayOfCycle = 0; // 1, 2, ...
```

Comment:
I'm not sure if we want to make dayOfCycle an inherent attribute of DataPoints. Could we make "on day x" calculations "on-the-fly" instead, but just going back from current day and see how many consecutive days of period-data we have? If a user's period starts on Monday, and then they don't get around to inputting data until Wednesday, they could input in the order of Wed->Tues->Mon, so calculating that number on instantiation isn't possible, and I'm not sure about the elegance of having the user enter the day #. LMK what you think

Reply:
The zero value is intended to be a "no value" until we are able to calculate this. I'm going to leave the property in here because it seems like calculating this on the fly every time they move to a new day is unnecessary calculation. Rather, we can leave it as the "no value" until we have new datapoints, where we extrapolate with the new data and re-assign this value. Will make this nullable though so we know that it's a "no value" more easily.